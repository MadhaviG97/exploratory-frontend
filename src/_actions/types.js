export const LOGIN_USER = "login_user";
export const REGISTER_USER = "register_user";
export const AUTH_USER = "auth_user";
export const LOGOUT_USER = "logout_user";
export const SEARCH = "search";
export const RENDER = "render";
export const COMMENTS = "comments";
export const REPLIES = "replies";
