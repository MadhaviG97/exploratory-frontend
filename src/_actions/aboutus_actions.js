import { ADD_MESSAGE } from "./types";

export const addMessage = async (messageData) => {
    const response = await fetch(`/aboutus/addmessage`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ ...messageData }),
    });
    const res = await response.json();
    return res.success;
  };