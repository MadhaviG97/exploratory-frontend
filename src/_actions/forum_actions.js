import {
  GET_QUESTIONS,
  ADD_QUESTION,
  ADD_ANSWER,
  GET_ANSWERS,
  EDIT_QUESTION,
  DELETE_QUESTION,
  RATE_QUESTION,
  EDIT_ANSWER,
  DELETE_ANSWER,
  RATE_ANSWER
} from "./types";

export const getQuestions = async () => {
  const response = await fetch(`/forum/questions`);
  const data = await response.json();
  const questions = data.data;
  return {
    type: GET_QUESTIONS,
    payload: questions,
  };
};

export const addQuestion = async (questionData) => {
  const response = await fetch(`/forum/addquestion`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...questionData }),
  });
  const res = await response.json();
  console.log(res);
  return {
    type: ADD_QUESTION,
    payload: questionData,
  };
};

export const getAnswers = async () => {
  const response = await fetch(`/forum/answers`);
  const data = await response.json();
  const answers = data.data;
  return {
    type: GET_ANSWERS,
    payload: answers,
  };
};

export const addAnswer = async (answerData) => {
  const response = await fetch(`/forum/addanswer`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...answerData }),
  });
  const res = await response.json();
  console.log(res);
  return {
    type: ADD_ANSWER,
    payload: answerData,
  };
};

export const deleteQuestion = async (questionData) => {
  const response = await fetch(`/forum/deletequestion`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...questionData }),
  });
  const res = await response.json();
  return {
    type: DELETE_QUESTION,
    payload: res,
  };
};

export const deleteAnswer = async (answerData) => {
  const response = await fetch(`/forum/deleteanswer`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...answerData }),
  });
  const res = await response.json();
  return {
    type: DELETE_ANSWER,
    payload: res,
  };
};

export const editQuestion = async (questionData) => {
  const response = await fetch(`/forum/editquestion`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...questionData }),
  });
  const res = await response.json();
  return {
    type: EDIT_QUESTION,
    payload: res,
  };
};

export const editAnswer = async (answerData) => {
  const response = await fetch(`/forum/editanswer`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...answerData }),
  });
  const res = await response.json();
  return {
    type: EDIT_ANSWER,
    payload: res,
  };
};