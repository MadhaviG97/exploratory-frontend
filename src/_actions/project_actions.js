import axios from "axios";
import { LOGIN_USER, SEARCH, RENDER } from "./types";

export function createResearch(dataToSubmit) {
  console.log(dataToSubmit);
  const request = axios
    .post(`/project/create-project`, dataToSubmit)
    .then((response) => response.data);
  return {
    type: LOGIN_USER,
    payload: request,
  };
}

export function search(dataToSubmit) {
  const request = axios
    .post(`/search`, dataToSubmit)
    .then((response) => response.data);

  return {
    type: SEARCH,
    payload: request,
  };
}

export function render(request) {
  return {
    type: RENDER,
    payload: request,
  };
}
