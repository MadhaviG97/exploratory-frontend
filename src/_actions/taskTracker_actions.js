import axios from "axios";
import {
  GET_TASKS,
  ADD_TASK,
  DELETE_TASK,
  EDIT_TASK,
  GET_TASK_COMMENTS,
  ADD_TASK_COMMENT,
  DELETE_TASK_COMMENT,
  EDIT_TASK_COMMENT,
  GET_PROJECT_COLLABORATORS
} from "./types";

export const getTasks = async (pId) => {
  const response = await fetch(`/project/tasktracker/tasks/${pId}`);
  const data = await response.json();
  const tasks = data.data;
  return {
    type: GET_TASKS,
    payload: tasks,
  };
};

export const addTask = async (taskData) => {
  const response = await fetch(`/project/tasktracker/addtask`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...taskData }),
  });
  const res = await response.json();
  console.log(res);
  return {
    type: ADD_TASK,
    payload: taskData,
  };
};

export const getComments = async (pId) => {
  const response = await fetch(`/project/tasktracker/comments/${pId}`);
  const data = await response.json();
  const comments = data.data;
  return {
    type: GET_TASK_COMMENTS,
    payload: comments,
  };
};

export const addComment = async (commentData) => {
  const response = await fetch(`/project/tasktracker/addcomment`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...commentData }),
  });
  const res = await response.json();
  console.log(res);
  return {
    type: ADD_TASK_COMMENT,
    payload: commentData,
  };
};

export const deleteTask = async (taskData) => {
  const response = await fetch(`/project/tasktracker/deletetask`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...taskData }),
  });
  const res = await response.json();
  return {
    type: DELETE_TASK,
    payload: res,
  };
};

export const deleteComment = async (commentData) => {
  const response = await fetch(`/project/tasktracker/deletecomment`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...commentData }),
  });
  const res = await response.json();
  return {
    type: DELETE_TASK_COMMENT,
    payload: res,
  };
};

export const editTask = async (taskData) => {
  const response = await fetch(`/project/tasktracker/edittask`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...taskData }),
  });
  const res = await response.json();
  return {
    type: EDIT_TASK,
    payload: res,
  };
};

export const editComment = async (commentData) => {
  const response = await fetch(`/project/tasktracker/editcomment`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...commentData }),
  });
  const res = await response.json();
  return {
    type: EDIT_TASK_COMMENT,
    payload: res,
  };
};

// export const getCollaborators = (pId) => {
//   const request = axios
//     .get(`/project/tasktracker/collaborators/${pId}`)
//     .then((response) => response.data.data);
//   return {
//     type: GET_PROJECT_COLLABORATORS,
//     payload: request,
//   };
// };

export const getCollaborators = async (pId) => {
  const response = await fetch(`/project/tasktracker/collaborators/${pId}`);
  const data = await response.json();
  const collaborators = data.data;
  return {
    type: GET_PROJECT_COLLABORATORS,
    payload: collaborators,
  };
};

