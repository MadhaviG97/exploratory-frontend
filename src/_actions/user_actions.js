import axios from 'axios';
import {
    LOGIN_USER,
    REGISTER_USER,
    AUTH_USER,
    LOGOUT_USER,
} from './types';


export function registerUser(dataToSubmit){
    const request = axios.post(`/register`,dataToSubmit)
        .then(response => response.data);
    
    return {
        type: REGISTER_USER,
        payload: request
    }
}

export function loginUser(dataToSubmit){
    const request = axios.post(`/login`,dataToSubmit)
                .then(response => {
                    if (response.message){

                    }else{
                        localStorage.setItem("token", response.data.token)
                        localStorage.setItem("body", response.data)
                        console.log(response.data.token)
                    }

                })
    

    return {
        type: LOGIN_USER,
        payload: request
    }
}

export function auth(){
    const token = localStorage.token;
    
    let config = {
        headers: {
        'Authorization': `Bearer ${token}`
        }
      }
    const request = axios.get(`/auth`,config)
    .then(response => response.data);
//if needed handle when token is invalid and add this code
            // if (data.message) {
            // An error will occur if the token is invalid.
            // If this happens, you may want to remove the invalid token.
            //localStorage.removeItem("token")
        //}
    return {
        type: AUTH_USER,
        payload: request
    }
}

export function logoutUser(){
    const request = axios.get(`/logout`)
    .then(response => response.data);

    return {
        type: LOGOUT_USER,
        payload: request
    }
}

