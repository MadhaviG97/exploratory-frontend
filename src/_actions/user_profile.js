import axios from "axios";
import {
  GET_PROFILE,
  UPDATE_PROFILE,
  GET_PROJECTS_BY_USER_ID,
  GET_PROJECT_POSTS_BY_USER_ID,
  GET_INSTITUTIONS,
  EDIT_PROFILE_PICTURE,
} from "./types";

export const getProfile = (id) => {
  const request = axios
    .get(`/userprofile/${id}`)
    .then((response) => response.data.data);
  return {
    type: GET_PROFILE,
    payload: request,
  };
};

export const editProfile = async (profileData) => {
  const response = await fetch(`/userprofile/edit`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...profileData }),
  });
  const res = await response.json();
  return {
    type: UPDATE_PROFILE,
    payload: res,
  };
};

export const editProfilePicture = async (profileData) => {
  const response = await fetch(`/userprofile/edit/profilepicture`, {
    method: "PATCH",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ ...profileData }),
  });
  const res = await response.json();
  return {
    type: EDIT_PROFILE_PICTURE,
    payload: profileData.url,
  };
};

export const getProjectsByUserId = async (uId) => {
  const response = await fetch(`/userprofile/projects/${uId}`);
  const data = await response.json();
  const projects = data.data;
  return {
    type: GET_PROJECTS_BY_USER_ID,
    payload: projects,
  };
};

export const getInstitutions = async () => {
  const response = await fetch(`/userprofile/edit/institutions`);
  const data = await response.json();
  const institutions = data.data;
  return {
    type: GET_INSTITUTIONS,
    payload: institutions,
  };
};

export const getProjectPostsByUserId = async (uId) => {
  const response = await fetch(`/userprofile/projects/posts/${uId}`);
  const data = await response.json();
  const projects = data.data;
  return {
    type: GET_PROJECT_POSTS_BY_USER_ID,
    payload: projects,
  };
};
