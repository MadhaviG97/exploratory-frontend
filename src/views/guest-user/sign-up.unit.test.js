import React from 'react';
import ReactDOM from 'react-dom';
import SignUp from './sign-up';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import * as actions from '../../_actions/user_actions'
import * as types from '../../_actions/types'
import fetchMock from 'fetch-mock'
const mockStore = configureMockStore([thunk]);
const store = mockStore({
    user: { userData: {isAuth:false} }
  });
jest.mock('react-router-dom', () => ({
    useHistory: () => ({
      push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
      }),
  }));
describe('signup', () => {
    afterEach(() => {
      fetchMock.reset()
      fetchMock.restore()
    })
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
        <Provider store={store}>
            <SignUp />
        </Provider>, div);
    });
    it('renders correctly', () => {
      const tree = renderer
        .create(
        <Provider store={store}>
          <SignUp />
        </Provider>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('creates REGISTER_USER when signup has been done', () => {
      fetchMock
        .getOnce('/register', { body: { email: 'my@gmail.com',password:'myname@1234',first_name:"my",last_name:"name",contact_no:"0978765768",confirm_password:"myname@1234" }, headers: { 'content-type': 'application/json' } })
  
      
      const expectedActions = [
        { type: types.REGISTER_USER, payload:  {
              "confirm_password": "myname@1234",
              "contact_no": "0978765768",
              "email": "my@gmail.com",
              "first_name": "my",
              "last_name": "name",
              "password": "myname@1234",
              }  }
      ]
      
      const store = mockStore({
          user: { userData: {isAuth:false} }
        })
      store.dispatch(actions.registerUser({ email: 'my@gmail.com',password:'myname@1234',first_name:"my",last_name:"name",contact_no:"0978765768",confirm_password:"myname@1234" }))
        // return of async actions
      expect(store.getActions()).toEqual(expectedActions)
        
    })
});
