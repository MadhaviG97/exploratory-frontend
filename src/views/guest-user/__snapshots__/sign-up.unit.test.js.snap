// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup renders correctly 1`] = `
<main
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
  />
  <div
    className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-4 MuiPaper-elevation6"
  >
    <div
      className="makeStyles-paper-3"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign-Up
      </h1>
      <form
        className="makeStyles-form-5"
        noValidate={true}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="firstName"
                id="firstName-label"
              >
                First Name
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="name"
                  autoFocus={true}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="firstName"
                  name="first_name"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-244"
                  >
                    <span>
                      First Name
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="lastName"
                id="lastName-label"
              >
                Last Name
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="name"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="lastName"
                  name="last_ame"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-244"
                  >
                    <span>
                      Last Name
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
                htmlFor="email"
                id="email-label"
              >
                Email Address
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="email"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  id="email"
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-244"
                  >
                    <span>
                      Email Address
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="Password"
                id="Password-label"
              >
                Password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  disabled={false}
                  id="Password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="password"
                  value=""
                />
                <div
                  className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="toggle password visibility"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-244"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="Confirm-Password"
                id="Confirm-Password-label"
              >
                Confirm Password
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  disabled={false}
                  id="Confirm-Password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="password"
                  value=""
                />
                <div
                  className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="toggle password visibility"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-244"
                  >
                    <span>
                      Confirm Password
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <br />
          <div
            className="makeStyles-wrapper-265"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                SIGN UP
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <br />
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="/signin"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Already have an account? Sign in
            </a>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
  />
</main>
`;
